module/zfs/zfs_chksum.c

```diff
@@ -50,9 +50,17 @@ typedef struct {
 	zio_checksum_tmpl_free_t *(free);
 } chksum_stat_t;
 
+#define	LIMIT_INIT	0
+#define	LIMIT_NEEDED	1
+#define	LIMIT_NOLIMIT	-1
+
+static int limit = LIMIT_INIT;
```

Comment:
```suggestion
static int chksum_stat_limit = LIMIT_INIT;
 ```

Or perhaps better yet you could avoid the global variable, move this in to `chksum_benchmark()`, and then add a flag to `chksum_benchit()` to request either the mini or full test run.  You could still early abort a full test run if any test too longer than some limit.

Reply:
I had it first within the function, but then the header stuff looked not so nice, so I decided to use two different header functions. But the 3x 0's and one headline are better for parsing - thats a point - I will switch back to my first version here.